beacon/resources/beacon.py

```diff
@@ -33,73 +34,73 @@ def main(request, response):
         id - the unique identifier of the test.
     """
 
-    id = request.GET.first("id")
-    command = request.GET.first("cmd").lower()
+    id = request.GET.first(b"id")
+    command = request.GET.first(b"cmd").lower()
 
     # Append CORS headers if needed.
-    if "origin" in request.GET:
-        response.headers.set("Access-Control-Allow-Origin",
-                             request.GET.first("origin"))
-    if "credentials" in request.GET:
-        response.headers.set("Access-Control-Allow-Credentials",
-                             request.GET.first("credentials"))
+    if b"origin" in request.GET:
+        response.headers.set(b"Access-Control-Allow-Origin",
+                             request.GET.first(b"origin"))
+    if b"credentials" in request.GET:
+        response.headers.set(b"Access-Control-Allow-Credentials",
+                             request.GET.first(b"credentials"))
 
     # Handle the 'store' and 'stat' commands.
-    if command == "store":
+    if command == b"store":
         error = None
 
         # Only store the actual POST requests, not any preflight/OPTIONS
         # requests we may get.
-        if request.method == "POST":
-            payload = ""
-            if "Content-Type" in request.headers and \
-               "form-data" in request.headers["Content-Type"]:
-                if "payload" in request.POST:
+        if request.method == u"POST":
+            payload = b""
+            if b"Content-Type" in request.headers and \
+               b"form-data" in request.headers[b"Content-Type"]:
+                if b"payload" in request.POST:
                     # The payload was sent as a FormData.
-                    payload = request.POST.first("payload")
+                    payload = request.POST.first(b"payload")
                 else:
                     # A FormData was sent with an empty payload.
                     pass
             else:
                 # The payload was sent as either a string, Blob, or BufferSource.
                 payload = request.body
 
-            payload_parts = filter(None, payload.split(":"))
+            payload_parts = list(filter(None, isomorphic_decode(payload).split(u":")))
```

Comment:
Can we not split a byte sequence? I'm not sure I see the need for decoding here.

Reply:
Do you prefer encoding `payload` into a text string then? Should the isomorphic encoding be used?

Reply:
Well when looking at the code it wasn't clear to me why we had to touch payload at all and could not treat it as a byte sequence.

Reply:
I have tried keeping "payload_parts" as byte sequence. One issue is that payload_parts[1][i] comes back as an integer (42 instead of '*') in python 3 case. Converting it to chr doesn't seem working for python 2. Still looking into the solution. Any suggestions are appreciated!

Reply:
A quick search suggests we could use slicing instead. So instead of payload_parts[1][i] we'd use payload_parts[1][i:i+1].

Reply:
Updated. Thank you!